modules/azurerm-virtual-network/variables.tf

```diff
@@ -0,0 +1,80 @@
+variable "address_space" {
+  description = "The address space that is used the virtual network. You can supply more than one address space."
+  type        = list(string)
+}
+
+variable "location" {
+  description = "The location/region where the virtual network is created. Changing this forces a new resource to be created."
+  type        = string
+}
+
+variable "name" {
+  description = "The name of the virtual network. Changing this forces a new resource to be created."
+  type        = string
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group in which to create the virtual network. Changing this forces a new resource to be created."
+  type        = string
+}
+
+variable "subnets" {
+  description = <<-EOF
+    [
+      {
+        address_prefixes: A list of address prefixes to use for the subnet
+        default_outbound_access_enabled: (Optional) Enable default outbound access to the internet for the subnet. Defaults to true.
+        name: The name of the subnet
+        route_table_id: (Optional) The ID of the Route Table that should be associated with this subnet.
+        security_group: (Optional) he Network Security Group to associate with the subnet. (Referenced by id, ie. azurerm_network_security_group.example.id)
+
+        deletegation: (Optional) One or more delegation blocks as defined below.
+      }
+    ]
+
+    delegation: {
+      name: A name for this delegation.
+      service_delegation: A service_delegation block as defined below.
+    }
+    
+    service_delegation: {
```

Comment:
It could be nice to link to the possible values for these, or add a note like _"See the documentation for the list of possible values."_

Reply:
Fair point, I'll add. Seems like `terraform-docs` does not preserve the indentation for heredocs as well, will need to clean this up a bit I think.

Reply:
Someone beat me to it as well, https://github.com/terraform-docs/terraform-docs/issues/683.

Reply:
Well, that's always nice validation!